src/reconcile/rec_write.c

```diff
@@ -868,6 +857,9 @@ __rec_split_chunk_init(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *
      *
      * In the case of fixed-length column-store, clear the entire buffer: fixed-length column-store
      * sets bits in bytes, where the bytes are assumed to initially be 0.
+     *
+     * FUTURE: pretty sure clearing the whole buffer is not necessary; also in any event the
+     * auxiliary space doesn't need to be cleared.
```

Comment:
Yeah. There are some wrinkles though; for one thing, in the common case most or all of the primary space is memcpy'd from the previous page image and then updated, so zeroing that part is completely pointless. Another is that if the bitsize isn't 8, there may be a partial byte at the end, and the unused part of it won't be changed by the bit accessors so we want to make sure it's zero. Plus any waste space needs to get zeroed so we don't write out random junk. I think the best way to deal with these bits is to zero them after writing the data, but I also need to see what that looks like in case it ends up interacting badly with splits. Anyhow, the follow-on ticket's already on my list. :-)

Reply:
Agreed.